KalturaWowzaServer/src/main/java/com/kaltura/media_server/services/Utils.java

```diff
@@ -231,10 +233,21 @@ public static String getStreamName(IMediaStream stream) {
     }
 
     public static String getMediaServerHostname() throws IOException, InterruptedException {
-        Process p = Runtime.getRuntime().exec("hostname -f");
+        return getMediaServerHostname(true);
+    }
+
+    public static String getMediaServerHostname(boolean full) throws IOException, InterruptedException {
+        if (mediaServerHostname != null)
```

Comment:
1.if the first one will request full=true  and the next will request full=false, then he will still get old one...
2.i think it's better to do it when loading the and not per request, since i rather not block the wowza thread for too long